locked-asset/lkmex-transfer/src/lib.rs

```diff
@@ -55,6 +63,50 @@ pub trait LkmexTransfer:
         self.add_energy_to_destination(receiver, &funds);
     }
 
+    #[endpoint(cancelTransfer)]
+    fn cancel_transfer(&self, sender: ManagedAddress, receiver: ManagedAddress) {
+        self.require_caller_has_admin_permissions();
+        let locked_funds_mapper = self.locked_funds(&receiver, &sender);
+        require!(!locked_funds_mapper.is_empty(), TRANSFER_NON_EXISTENT);
+
+        let locked_funds = locked_funds_mapper.get();
+        locked_funds_mapper.clear();
+        self.all_senders(&receiver).swap_remove(&sender);
+        self.address_last_transfer_epoch(&sender).clear();
+        self.transfer_blacklist().add(&receiver);
+
+        self.send().direct_multi(&sender, &locked_funds.funds);
+        self.add_energy_to_destination(sender, &locked_funds.funds);
+    }
+
+    #[endpoint(addAddressToBlacklist)]
+    fn add_address_to_blacklist(&self, address: ManagedAddress) {
+        self.require_caller_has_admin_permissions();
+        let addresses_mapper = self.transfer_blacklist();
+        require!(!addresses_mapper.contains(&address), ADDRESS_BLACKLISTED);
+        addresses_mapper.add(&address);
+    }
```

Comment:
I don't think we need manual blacklist adding, since it's already integrated inside the cancel_transfer logic.

Reply:
If you think this is indeed needed, then please also apply the suggestions from the `remove` endpoint below.

Reply:
Removed.